### PR TITLE
feat: Add panel for creating questions

### PR DESCRIPTION
**Pull request by pmrukot**
**Summary**

This PR introduces a new view called "Panel". At first, it will support adding questions to existing categories.

screen shot 2017-07-08 at 14 27 53

TODO:
- [x] Create submit form in elm
- [x] Integrate with backend

**Related issues**
https://github.com/pmrukot/aion/issues/42

### CODE REVIEW

**Comment by mrapacz** [comment]
Looks good so far, good luck on the way!

File: aion/web/elm/src/Panel/View.elm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+panelView : Model -> Html Msg
+panelView model =
+    div []
+        [ h3 [] [ text "Create new question for certain category:" ]
+        , form []
+            [ p [] [ text "Enter question content below:" ]
+            , input [ placeholder "Zawsze spada na 4 łapy", onInput SetNewQuestionContent ] []
+            , p [] [ text "Enter answer below, separate all posibilities with comma:" ]
+            , input [ placeholder "Kot,kocur,kotek", onInput SetNewAnswerContent ] []
```
Comment (mrapacz) [inline aion/web/elm/src/Panel/View.elm]: I wouldn't mix up Polish with English here, maybe change these to whatever in English, just so we stay cohesive

File: aion/web/elm/src/Update.elm [hunk 2]
```diff
@@ -114,3 +123,39 @@ update msg model =
...
+
+        SetNewAnswerContent answerContent ->
+            let
+                oldPanelData =
+                    model.panelData
+
+                newPanelData =
+                    { oldPanelData | newAnswerContent = answerContent }
+            in
+                { model | panelData = newPanelData } ! []
```
Comment (mrapacz) [inline aion/web/elm/src/Update.elm]: Actually these can be rewritten in a simpler manner:
```        SetNewAnswerContent answerContent ->
            let
                oldPanelData =
                    model.panelData
            in
                { model | panelData = { oldPanelData | newAnswerContent = answerContent } } ! []```

**Review by mrapacz — CHANGES_REQUESTED** [review]
Maybe we should think about some form or precaution against submitting the same form a couple of times? We could either not digest the same request twice or display a notification after submitting the first correct request from Elm. Anyway, what happens now if the input cannot be inserted into db? The error msgs are only local, the user won't see them. 

Apart from that, the code looks pretty neat, tell me what you think on the precautions (we can add it in another PR possibly?) and fix the smaller things I marked and I think we can safely pull it :+1: :100: 

One last side note, a simple UX thing - 
image
Maybe we should get rid of the quotation marks here?

File: aion/web/models/question.ex [hunk 3]
```diff
@@ -13,8 +15,17 @@ defmodule Aion.Question do
   Builds a changeset based on the `struct` and `params`.
   """
   def changeset(struct, params \\ %{}) do
-    struct
-    |> cast(params, [:content, :image_name])
-    |> validate_required([:content, :image_name])
+    if params["subject"] do
+      subject = Repo.get(Subject, params["subject"])
+
+      struct
```
Comment (jtkpiotr) [inline aion/web/models/question.ex]: Maybe we can create separate methods for creating changeset for params that contains and does not contain subject.

Comment (pmrukot) [reply aion/web/models/question.ex]: I refactored it to be similar to answer's changeset and changed controller a little to better handle different cases.

**Comment by pmrukot** [comment]
@mrapacz
About quotation marks: currently we are using `elm-select` which uses types as select options. I am giving him list of strings though, not type. And there is currently no support in Elm for function like haskell's well known [`read`](http://zvon.org/other/haskell/Outputprelude/read_f.html). No sure how can we handle it.

I think that we should merge it and reiterate to make it user friendly. Purpose of this PR was to make end-to-end Question+Answer creation, with simple form. Let's assume happy path since we are in early stage of development and add improvements for user experience in next PR. I can open an issue but I suggest following improvements:

- [ ] Possibility to add more than one possible answer
- [ ] Notifications if the request was successful or not
- [ ] Form validation
- [ ] Clearing the form after successful submit

Any more?

File: aion/web/elm/src/Panel/View.elm [hunk 4]
```diff
@@ -16,9 +16,9 @@ panelView model =
         [ h3 [] [ text "Create new question for certain category:" ]
         , form []
             [ p [] [ text "Enter question content below:" ]
-            , input [ placeholder "Zawsze spada na 4 łapy", onInput SetNewQuestionContent ] []
+            , input [ placeholder "How much is 2+2?", onInput SetNewQuestionContent ] []
```
Comment (mrapacz) [inline aion/web/elm/src/Panel/View.elm]: 👍 💯 🥇

**Comment by mrapacz** [comment]
Okay, we can do it in other PRs, although if we want functionality first, then we should start with adding multiple answers first, then move to UX and similar

**Comment by mrapacz** [comment]
retrokid

**Review by mrapacz — APPROVED** [review]

**Review by jtkpiotr — APPROVED** [review]

**Pull request merged**